editor/import/3d/resource_importer_scene.cpp

```diff
@@ -3075,7 +3075,11 @@ Error ResourceImporterScene::import(ResourceUID::ID p_source_id, const String &p
 		}
 	}
 
-	scene = _generate_meshes(scene, mesh_data, gen_lods, create_shadow_meshes, LightBakeMode(light_bake_mode), lightmap_texel_size, src_lightmap_cache, mesh_lightmap_caches);
+	try {
+		scene = _generate_meshes(scene, mesh_data, gen_lods, create_shadow_meshes, LightBakeMode(light_bake_mode), lightmap_texel_size, src_lightmap_cache, mesh_lightmap_caches);
+	} catch (const std::runtime_error &e) {
+		ERR_FAIL_V_EDMSG(ERR_FILE_CORRUPT, "Bad array access while generating mesh. The imported scene is corrupt.");
+	}
```

Comment:
As STL is not used, try/catch aren't also, as we try to stick most of the time to C-like code.

It's not explicitly said here, but this is a good read nonetheless.

https://docs.godotengine.org/en/stable/contributing/development/cpp_usage_guidelines.html